itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/txn/compactor/CompactionPoolOnTezTest.java

```diff
@@ -171,4 +174,30 @@ public void testShowCompactionsRespectPoolName() throws Exception {
     Assert.assertTrue(p.matcher(results.get(1).toString()).matches());
   }
 
+  @Test
+  public void testCompactionWithCustomPool() throws Exception {
+    conf.setInt("hive.compactor.worker.pool1.threads", 1);
+
+    Map<String, String> properties = new HashMap<>();
+    properties.put(Constants.HIVE_COMPACTOR_WORKER_POOL, "pool1");
+    provider.createFullAcidTable(null, DEFAULT_TABLE_NAME, false, false, properties);
+    provider.insertTestData(DEFAULT_TABLE_NAME, false);
+
+    TxnCommandsBaseForTests.runInitiator(conf);
+
+    checkCompactionRequest("initiated", "pool1");
+
+    Map<String, Integer> customPools = CompactorUtil.getPoolConf(conf);
+    Assert.assertEquals(customPools.size(), 1);
+    Assert.assertEquals(customPools.get("pool1"), new Integer(1));
+    
+    Worker worker = new Worker();
```

Comment:
could you use helper method: `TxnCommandsBaseForTests.runWorker(conf)`?

Reply:
Done